modules/core/include/frame.h

```diff
@@ -90,6 +98,8 @@ class Frame{
      */
     void save(std::string dir);
 
+    void undistort(Camera &camera);
```

Comment:
Neither, you pass the camera to the created frame and it handles it from there. 
Frame::undistort is for testing purposes since it allows you to use a different camera than when the Frame was first created. 
Admittedly this could be more clear, I'll add some clarification on that point.